src/renderer/kube-object.store.ts

```diff
@@ -11,14 +11,25 @@ import { getHostedCluster } from "../common/cluster-store";
 @autobind()
 export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemStore<T> {
   abstract api: KubeApi<T>;
-  public limit: number;
+  public readonly limit?: number;
+  public readonly bufferSize: number = 50000;
 
   constructor() {
     super();
     this.bindWatchEventsUpdater();
     kubeWatchApi.addListener(this, this.onWatchApiEvent);
   }
 
+  get query(): IKubeApiQueryParams {
+    const { limit } = this;
+
+    if (!limit) {
```

Comment:
So `{limit: 0}` doesn't make any sense?

Reply:
This was just a direct move from the old line 66 below. I don't know... does kube treat `limit: 0` as the same as not passing `limit` at all? (given than kube is written in Go, that is probably the case sadly).